iaso/api/payments/views.py

```diff
@@ -0,0 +1,156 @@
+        ).distinct()
+
+        return queryset
+
+    @swagger_auto_schema(auto_schema=None)
+    def retrieve(self, request, *args, **kwargs):
+        raise NotFound("Retrieve operation is not allowed.")
+
+    @swagger_auto_schema(
+        manual_parameters=[
+            openapi.Parameter(
+                name="users",
+                in_=openapi.IN_QUERY,
+                description="A comma-separated list of User IDs associated with the payments",
+                type=openapi.TYPE_STRING,
+            ),
+            openapi.Parameter(
+                name="user_roles",
+                in_=openapi.IN_QUERY,
+                description="A comma-separated list of User Role IDs associated with the payments",
+                type=openapi.TYPE_STRING,
+            ),
+            openapi.Parameter(
+                name="parent_id",
+                in_=openapi.IN_QUERY,
+                description="The ID of the parent organization unit linked to the change requests. This should also include child units.",
+                type=openapi.TYPE_INTEGER,
+            ),
+            openapi.Parameter(
+                name="change_requests__created_at_after",
+                in_=openapi.IN_QUERY,
+                description="The start date for when the change request has been validated. Format: YYYY-MM-DD",
+                type=openapi.TYPE_STRING,
+                format=openapi.FORMAT_DATE,
+            ),
+            openapi.Parameter(
+                name="change_requests__created_at_before",
+                in_=openapi.IN_QUERY,
+                description="The end date for when the change request has been validated. Format: YYYY-MM-DD",
+                type=openapi.TYPE_STRING,
+                format=openapi.FORMAT_DATE,
+            ),
+        ]
+    )
+    def list(self, request):
+        orders = request.GET.get("order", "user__last_name").split(",")
+
+        users_with_change_requests = (
+            OrgUnitChangeRequest.objects.filter(status=OrgUnitChangeRequest.Statuses.APPROVED)
+            .values("created_by")
+            .annotate(num_requests=Count("created_by"))
+            .filter(num_requests__gt=0)
+        )
+
+        for user in users_with_change_requests:
+            change_requests = OrgUnitChangeRequest.objects.filter(
+                created_by_id=user["created_by"],
+                status=OrgUnitChangeRequest.Statuses.APPROVED,
+            )
+            if change_requests.exists():
+                potential_payment, created = PotentialPayment.objects.get_or_create(
+                    user_id=user["created_by"],
+                )
+                for change_request in change_requests:
+                    if not Payment.objects.filter(change_requests__id=change_request.id).exists():
+                        potential_payment.change_requests.add(change_request)
+                potential_payment.save()
+        queryset = self.filter_queryset(self.get_queryset()).order_by(*orders)
```

Comment:
We need to discuss about using this and also `filters.OrderingFilter` and `ordering_fields`